### PR TITLE
TIP-409: refactor completeness panel

### PR DESCRIPTION
**Pull request by juliensnz**
We cleaned the completeness code using a panel decorator and removing unused code.

### CODE REVIEW

File: features/product/completeness/display_completeness.feature [hunk 1]
```diff
@@ -76,11 +76,11 @@ Feature: Display the completeness of a product
...
     And I should see the completeness:
-      | channel | locale | state   | missing_values                               | ratio |
-      | mobile  | en_US  | warning | name price size                              | 40%   |
-      | mobile  | fr_FR  | warning | price size                                   | 60%   |
-      | tablet  | en_US  | warning | name description price rating side_view size | 25%   |
-      | tablet  | fr_FR  | warning | price rating side_view size                  | 50%   |
+      | channel | locale | state   | missing_values                                   | ratio |
+      | mobile  | en_US  | warning | Name, Price, Size                                | 40%   |
+      | mobile  | fr_FR  | warning | Price, Size                                      | 60%   |
+      | tablet  | en_US  | warning | Name, Description, Price rating, Side view, Size | 25%   |
```
Comment (willy-ahva) [inline features/product/completeness/display_completeness.feature]: Rating

File: features/product/completeness/display_completeness.feature [hunk 2]
```diff
@@ -76,11 +76,11 @@ Feature: Display the completeness of a product
...
+      | tablet  | fr_FR  | warning | Price, rating, Side view, Size                   | 50%   |
```
Comment (willy-ahva) [inline features/product/completeness/display_completeness.feature]: Rating

File: features/product/completeness/ensure_display_for_locale_specific.feature [hunk 3]
```diff
@@ -29,55 +29,54 @@ Feature: Proper completeness display for locale specific attributes
...
-    Then I should see the missing completeness labels:
-      | channel   | locale | missing_value   | expected_label    |
-      | ecommerce | de_DE  | description     | Beschreibung      |
-      | ecommerce | fr_FR  | description     | Beschreibung      |
-      | ecommerce | de_DE  | thumbnail       | Miniaturansicht   |
-      | ecommerce | fr_FR  | thumbnail       | Miniaturansicht   |
-      | ecommerce | de_DE  | legend          | Legende           |
-      | ecommerce | fr_FR  | legend          | Legende           |
-      | ecommerce | fr_FR  | locale_specific | [locale_specific] |
+    And I should see the completeness:
```
Comment (willy-ahva) [inline features/product/completeness/ensure_display_for_locale_specific.feature]: Then

File: features/Pim/Behat/Context/Domain/Enrich/CompletenessContext.php [hunk 4]
```diff
@@ -117,7 +116,34 @@ public function iShouldSeeTheCompleteness(TableNode $table)
...
+            ), true)
         ));
     }
+
+    /**
+     * Display the completeness from html in flat format
+     *
+     * @param array $structuredCompleteness
+     *
+     * @return array
```
Comment (juliensnz) [inline features/Pim/Behat/Context/Domain/Enrich/CompletenessContext.php]: add doc for input/output

File: features/Pim/Behat/Context/Domain/Enrich/CompletenessContext.php [hunk 5]
```diff
@@ -87,10 +87,9 @@ public function theCompletenessShouldBeOpenedOrClosed($localeCode, $state)
...
         $table             = $table->getHash();
-        $completenessPanel = $this->getCurrentPage()->getElement('Completeness');
 
-        $this->spin(function () use ($table, $completenessPanel) {
-            $completenessData = $completenessPanel->getCompletenessData();
+        $this->spin(function () use ($table) {
+            $completenessData = $this->getCurrentPage()->getElement('Completeness')->getCompletenessData();
 
             foreach ($table as $expected) {
                 $scopeBlock = $completenessData[$expected['locale']]['data'][$expected['channel']];
```
Comment (juliensnz) [inline features/Pim/Behat/Context/Domain/Enrich/CompletenessContext.php]: use phpspec diff

File: features/Pim/Behat/Context/Domain/Enrich/CompletenessContext.php [hunk 6]
```diff
@@ -14,49 +16,134 @@
...
+    /**
+     * @param TableNode $table
+     *
+     * @Then /^I should see the completeness:$/
+     *
+     * @throws ExpectationException
+     */
+    public function iShouldSeeTheCompleteness(TableNode $table)
+    {
+        $table             = $table->getHash();
```
Comment (grena) [inline features/Pim/Behat/Context/Domain/Enrich/CompletenessContext.php]: extra spaces

File: features/Pim/Behat/Context/Domain/Enrich/CompletenessContext.php (hunk 4 above)
Comment (phaseinducer) [inline features/Pim/Behat/Context/Domain/Enrich/CompletenessContext.php]: done

**Pull request merged**